CRM/Contact/Form/Task/PDFLetterCommon.php

```diff
@@ -363,7 +365,12 @@ public static function postProcess(&$form) {
 
     // CRM-16725 Skip creation of activities if user is previewing their PDF letter(s)
     if ($buttonName == '_qf_PDF_upload') {
-      $activityIds = self::createActivities($form, $html_message, $form->_contactIds);
+
+      // This seems silly, but the old behavior was to first check `_cid`
+      // and then use the provided `$contactIds`. Probably not even necessary,
+      // but difficult to audit.
+      $contactIds = $form->_cid ? array($form->_cid) : $form->_contactIds;
```

Comment:
I've moved this weirdness out of the shared function to each form so we can work to remove it - impossible to ever get rid of it, per comments, if hidden in a shared function